ctx.go

```diff
@@ -95,7 +95,7 @@ func NewCtxWithVersion(version SSLVersion) (*Ctx, error) {
 	case TLSv1_2:
 		method = C.X_TLSv1_2_method()
 	case AnyVersion:
-		method = C.X_SSLv23_method()
+		method = C.X_TLS_method()
```

Comment:
Why is here no case for `TLSv1_3`?

Reply:
Using the method parameter is deprecated in openssl now, so there is no TLSv1_3. That's why I added the min and max proto functions.

Reply:
I think we'll need to build an abstraction around that then, i.e. we shouldn't have a `NewCtxWithVersion` **and** a `Set{Min,Max}ProtoVersion` at the same time.
I'd be ok with either:
* removing `NewCtxWithVersion`, and forcing users to use `Set{Min,Max}ProtoVersion`, or
* not introducing `Set{Min,Max}ProtoVersion` as a public function, but making it private and then calling it from `NewCtxWithVersion`

The first option probably allows for more flexibility (as you can set min and max separately), whereas `NewCtxWithVersion` forces you to use min = max, right?

What do you think?

Reply:
I think the real question is what is the policy for deprecated functions in OpenSSL?

I've seen in some other Go wrapper libraries that by default they don't include deprecated functions but have a tag to enable them at compile time. I can see implementing something before the next version, but that's a PR in itself.

Reply:
As a general rule, I'd try to avoid deprecated functions, if possible. After all, they're deprecated for a reason.

My point in my previous comment was that we should abstract away that complexity from users of `go-openssl`. That means, we should **either** offer `NewCtxWithVersion` **or** `Set{Min,Max}ProtoVersion`.